src/presentation/cli/controller/services.go

```diff
@@ -0,0 +1,274 @@
+		&portBindingsSlice, "port-bindings", "p", []string{}, "PortBindings (port/protocol)",
+	)
+	cmd.Flags().IntVarP(
+		&timeoutStartSecsInt, "timeout-start-secs", "o", 0, "TimeoutStartSecs",
+	)
+	cmd.Flags().BoolVarP(
+		&autoStart, "auto-start", "s", true, "AutoStart",
+	)
+	cmd.Flags().IntVarP(
+		&maxStartRetriesInt, "max-start-retries", "m", 0, "MaxStartRetries",
+	)
+	cmd.Flags().BoolVarP(
+		&autoRestart, "auto-restart", "r", true, "AutoRestart",
+	)
+	cmd.Flags().BoolVarP(
+		&autoCreateMapping, "auto-create-mapping", "a", true, "AutoCreateMapping",
+	)
+	return cmd
+}
+
+func (controller *ServicesController) CreateCustom() *cobra.Command {
+	var nameStr, typeStr, startCmdStr, versionStr string
+	var envsSlice, portBindingsSlice []string
+	var autoStart, autoRestart, autoCreateMapping bool
+	var timeoutStartSecsInt, maxStartRetriesInt int
+
+	cmd := &cobra.Command{
+		Use:   "create-custom",
+		Short: "CreateCustomService",
+		Run: func(cmd *cobra.Command, args []string) {
+			requestBody := map[string]interface{}{
+				"name":              nameStr,
+				"type":              typeStr,
+				"startCmd":          startCmdStr,
+				"autoStart":         autoStart,
+				"autoRestart":       autoRestart,
+				"autoCreateMapping": autoCreateMapping,
+			}
+
+			if len(envsSlice) > 0 {
+				requestBody["envs"] = envsSlice
+			}
+
+			if versionStr != "" {
+				requestBody["version"] = versionStr
+			}
+
+			if len(portBindingsSlice) > 0 {
+				requestBody["portBindings"] = portBindingsSlice
+			}
+
+			if timeoutStartSecsInt != 0 {
+				requestBody["timeoutStartSecs"] = uint(timeoutStartSecsInt)
+			}
+
+			if maxStartRetriesInt != 0 {
+				requestBody["maxStartRetries"] = uint(maxStartRetriesInt)
+			}
+
+			cliHelper.ServiceResponseWrapper(
+				controller.servicesService.CreateCustom(requestBody),
+			)
+		},
+	}
+
+	cmd.Flags().StringVarP(&nameStr, "name", "n", "", "ServiceName")
+	cmd.MarkFlagRequired("name")
+	cmd.Flags().StringVarP(&typeStr, "type", "t", "", "ServiceType (application|database|runtime|other)")
+	cmd.MarkFlagRequired("type")
+	cmd.Flags().StringVarP(&startCmdStr, "start-command", "c", "", "StartCommand")
+	cmd.MarkFlagRequired("start-command")
+	cmd.Flags().StringSliceVarP(&envsSlice, "envs", "e", []string{}, "Envs (name=value)")
+	cmd.Flags().StringVarP(&versionStr, "version", "v", "", "ServiceVersion")
+	cmd.Flags().StringSliceVarP(
+		&portBindingsSlice, "port-bindings", "p", []string{}, "PortBindings (port/protocol)",
+	)
+	cmd.Flags().IntVarP(
+		&timeoutStartSecsInt, "timeout-start-secs", "o", 0, "TimeoutStartSecs",
+	)
+	cmd.Flags().BoolVarP(
+		&autoStart, "auto-start", "s", true, "AutoStart",
+	)
+	cmd.Flags().IntVarP(
+		&maxStartRetriesInt, "max-start-retries", "m", 0, "MaxStartRetries",
+	)
+	cmd.Flags().BoolVarP(
+		&autoRestart, "auto-restart", "r", true, "AutoRestart",
+	)
+	cmd.Flags().BoolVarP(
+		&autoCreateMapping,
+		"auto-create-mapping",
+		"a",
+		true,
+		"AutoCreateMapping",
```

Comment:
Too much line breaks. Please check the entire code for the same errors I report, I won't be mention where you made the same mistake twice or more.

Reply:
Done!